### DISCUSSION TITLE
is there any way to achieve zero copy deserialization

### DISCUSSION THREAD

**Discussion by Hydrostic** [ntex-rs/ntex#291] [Q&A]
In the framework I used previously, I could lazily extract data from the body using the builtin method. However, in ntex, there seems to be no way to achieve this simply. I wonder if ntex provides such a method or do I have to implement it myself?

**Comment by fafhrd91** [comment]
could you explain what is “lazily extract”?

**Comment by Hydrostic** [comment]
i.e. extract in the handler. But actually it seems I have mistaken the idea of zero copy deserialzation, actually I want some method that bind the lifetime to `Deseralize<'de>` that can help to write code like below:
```
pub struct Data<'a>{
   text: Cow<'a, str>
}
```

**Comment by fafhrd91** [comment]
you need to load body in any case, then you can do whatever you need

**Comment by Hydrostic** [comment]
But why the builtin extractor is designed to only support self-owned struct, I wonder if changing the generic lifetime to below would be a better approach
```rust
impl<'de, T, Err: ErrorRenderer> FromRequest<Err> for Json<T>
where
    T: Deserialize<'de>
{
```

**Comment by fafhrd91** [comment]
lifetime would complicate definition

**Comment by fafhrd91** [comment]
you can always take Payload and use whatever extraction method you need

**Comment by Hydrostic** [comment]
Ok, Thanks
